src/webhook/test_namespace_validation.py

```diff
@@ -84,33 +90,55 @@ def runtest(self, namespace, groups, expect):
                 failmsg)
 
     def test_deny(self):
-        # In order to get DENYs, we must have *both* a privileged namespace
-        # *and* the 'dedicated-admins' group.
-        for ns in PRIVILEGED_NAMESPACES:
-            for gl in GROUP_LISTS:
-                # Always include dedicated-admins
-                groups = gl + ('dedicated-admins',)
-                self.runtest(ns, groups, False)
+        # layered-cs-sre-admins should not have access to any privledged namespaces except ones starting with 'redhat'
+        # dedicated-admins should not have access to any privledged namespaces
+        for ns in self.PRIVILEGED_NAMESPACES:
+            for user in self.NON_PRIVILEGED_USERS:
+                groups = ('layered-cs-sre-admins',)
+                self.runtest(ns, groups, user, False)
+                groups = ('dedicated-admins',)
+                self.runtest(ns, groups, user, False)
+                groups = ('random-test-group',)
+                self.runtest(ns, groups, user, False)
 
     def test_allow_group(self):
-        # If the group list doesn't contain 'dedicated-admins', always ALLOW,
-        # even if the namespace is privileged. (Really?)
-        for ns in PRIVILEGED_NAMESPACES + NONPRIV_NAMESPACES:
-            for gl in GROUP_LISTS:
-                self.runtest(ns, gl, True)
+        # user in dedicated-admins group can edit non-privileged NS
+        # user in osd-sre-admins group can edit non-privileged NS
+        # user in osd-sre-admins group can edit privileged NS
+        for ns in self.PRIVILEGED_NAMESPACES + self.NONPRIV_NAMESPACES + self.REDHAT_NAMESPACES:
+            for user in self.PRIVILEGED_USERS:
```

Comment:
you have too many variables going on in these test cases for these tests to provide meaningful value. in these test cases, you are effectively making sure that kube:admin and system:admin in groups osd-sre-cluster-admins or osd-sre-admins are able to edit all namespaces. you need to separate these two cases out: 1) kube:admin and system:admin should be able to edit all NS, even if they are in some non-privileged group ("random-test-group"); and 2) NON_PRIVILEGED_USERS should be able to edit all NS if they are in either the osd-sre-cluster-admins or osd-sre-admins group. case 1 is on lines 122-128, so let's just change line 109:
```suggestion
            for user in self.NON_PRIVILEGED_USERS:
```